Test/domains/urcu-bp.hpp

```diff
@@ -3,22 +3,23 @@
 #include "rcu_domain.hpp"
 
 #include <urcu-bp.h>
-#define _LGPL_SOURCE
 
-namespace std {
-    class rcu_bp: public rcu_domain {
-    public:
-        void register_thread() { rcu_register_thread(); }
-        void unregister_thread() { rcu_unregister_thread(); }
-        void read_lock() noexcept { rcu_read_lock(); }
-        void read_unlock() noexcept { rcu_read_unlock(); }
-        void synchronize() noexcept { synchronize_rcu(); }
-        void retire(rcu_head *rhp, void (*cbf)(rcu_head *rhp)) { call_rcu(rhp, cbf); }
-        void barrier() noexcept { rcu_barrier(); }
-        void quiescent_state() noexcept { rcu_quiescent_state(); }
-        void thread_offline() noexcept { rcu_thread_offline(); }
-        void thread_online() noexcept { rcu_thread_online(); }
+class rcu_domain_bp {
+public:
+    static constexpr bool register_thread_needed() { return false; }
+    void register_thread() { rcu_register_thread(); }
+    void unregister_thread() { rcu_unregister_thread(); }
+    void thread_offline() noexcept { rcu_thread_offline(); }
+    void thread_online() noexcept { rcu_thread_online(); }
 
-        static constexpr bool register_thread_needed() { return false; }
-    };
-} // namespace std
+    static constexpr bool quiescent_state_needed() { return false; }
+    void quiescent_state() noexcept { rcu_quiescent_state(); }
```

Comment:
The big requirement from my viewpoint is that a program that doesn't invoke quiescent_state can get warnings if it tries to use rcu_domain_qsbr.  It seems like that can be made to work with either of  your alternatives, for example, having an RcuDomainsQsbr concept (or similar).

From a pure C++ viewpoint, which is best?